clang/test/CIR/CodeGen/derived-to-base.cpp

```diff
@@ -77,6 +77,9 @@ void C3::Layer::Initialize() {
 
 // CHECK-DAG: !ty_22C23A3ALayer22 = !cir.struct<class "C2::Layer"
 // CHECK-DAG: !ty_22C33A3ALayer22 = !cir.struct<struct "C3::Layer"
+// CHECK-DAG: !ty_22A22 = !cir.struct<class "A" {!cir.ptr<!cir.ptr<!cir.func<!cir.int<u, 32> ()>>>, !cir.int<s, 32>}>
+// CHECK-DAG: !ty_22A2Ebase22 = !cir.struct<class "A.base" {!cir.ptr<!cir.ptr<!cir.func<!cir.int<u, 32> ()>>>, !cir.int<s, 32>} #cir.record.decl.ast>
```

Comment:
> In the original codegen a new type is created for the base class

It seems like the new `A.base` class is nearly identical to the original `A` class. Wouldn't it be possible to add the `#cir.record.decl.ast` attribute in class `A` and reuse it instead?

Reply:
well, probably yes!